### PR TITLE
[NO-TASK] Add assignTransactions method

### PR DESCRIPTION
**Pull request by Kolibri1990**
#### Pre-Release

- [ ] Check migrations
  - No database related infos (sqldb-xxx)
  - Impact on GS (new/removed columns)
- [ ] Check for linter errors (in PR)
- [ ] Test basic user operations
  - Login/logout
  - Registration
  - Update user data
  - Create/delete payment routes

#### Post-Release

- Test basic user operations
- Monitor application insights log

### CODE REVIEW

File: src/subdomains/supporting/bank/bank-tx/bank-tx.service.ts [hunk 1]
```diff
@@ -66,6 +73,19 @@ export class BankTxService {
     if (olkyTransactions.length > 0) await this.settingService.set(settingKeyOlky, newModificationTime);
   }
 
+  async assignTransactions() {
+    const unassignedBankTx = await this.bankTxRepo.find({ where: { type: IsNull() } });
+
+    for (const tx of unassignedBankTx) {
+      const match = /^[0-9A-Z]{4}-[0-9A-Z]{4}-[0-9A-Z]{4}$/.test(tx.remittanceInfo);
```
Comment (davidleomay) [inline src/subdomains/supporting/bank/bank-tx/bank-tx.service.ts:80]: Put the regex to `Config.formats`

File: src/subdomains/supporting/bank/bank-tx/bank-tx.service.ts [hunk 2]
```diff
@@ -66,6 +73,19 @@ export class BankTxService {
...
+
+      if (match) {
+        const buy = await this.buyService.getBuyRepo().findOneBy({ bankUsage: tx.remittanceInfo });
+        if (buy) await this.update(tx.id, { type: BankTxType.BUY_CRYPTO, buyId: buy.id });
```
Comment (davidleomay) [inline src/subdomains/supporting/bank/bank-tx/bank-tx.service.ts:84]: Maybe add a `getByBankUsage` method to buy service and remove the `getBuyRepo`?

Comment (davidleomay) [reply src/subdomains/supporting/bank/bank-tx/bank-tx.service.ts:84]: You need to find by the regex match. Means you need to remove the start/end conditions on the regex and adapt the check to
```ts
const match = /[0-9A-Z]{4}-[0-9A-Z]{4}-[0-9A-Z]{4}/.exec(tx.remittanceInfo);
if (match) {
   const buy = await this.buyService.getBuyRepo().findOneBy({ bankUsage: match[0] });
   ...
}
```

File: src/config/config.ts [hunk 3]
```diff
@@ -70,6 +70,7 @@ export class Configuration {
     signature: /^(.{87}=|[a-f0-9]{130}|[a-f0-9x]{132}|[a-f0-9]{582}|[a-z0-9]{104}|[a-z0-9]{140,146})$/,
     key: /^[a-f0-9]{84}$/,
     ref: /^(\w{1,3}-\w{1,3})$/,
+    bankUsage: /^[0-9A-Z]{4}-[0-9A-Z]{4}-[0-9A-Z]{4}$/,
```
Comment (davidleomay) [inline src/config/config.ts:73]: Remove start/end conditions `/[0-9A-Z]{4}-[0-9A-Z]{4}-[0-9A-Z]{4}/`

File: src/subdomains/supporting/bank/bank-tx/bank-tx.service.ts [hunk 4]
```diff
@@ -34,11 +35,17 @@ export class BankTxService {
     private readonly olkyService: OlkypayService,
     private readonly bankTxReturnService: BankTxReturnService,
     private readonly bankTxRepeatService: BankTxRepeatService,
+    private readonly buyService: BuyService,
   ) {}
 
   // --- TRANSACTION HANDLING --- //
   @Cron(CronExpression.EVERY_MINUTE)
```
Comment (davidleomay) [inline src/subdomains/supporting/bank/bank-tx/bank-tx.service.ts:42]: Decrease to every 30s (requested by @DFXAdmin 😉)

**Review by davidleomay — APPROVED** [review]

File: src/subdomains/supporting/bank/bank-tx/bank-tx.service.ts (hunk 4 above)
Comment (DFXAdmin) [reply src/subdomains/supporting/bank/bank-tx/bank-tx.service.ts:42]: Bisch de bescht 😀

**Pull request merged**